tests/experimental/pydantic/test_conversion.py

```diff
@@ -645,13 +661,153 @@ class UserInput:
+
+    data = UserInput(age=1)
+    user = data.to_pydantic()
+
+    assert user.age == 1
+    assert user.password is None
+
+    definition: TypeDefinition = UserInput._type_definition
+    assert definition.name == "UserInput"
+
+    [
+        age_field,
+        password_field,
+    ] = (
+        definition.fields
+    )  # fields without a default go first, so the order gets reverse
+
+    assert age_field.python_name == "age"
+    assert age_field.type is int
+
+    assert password_field.python_name == "password"
+    assert isinstance(password_field.type, StrawberryOptional)
+    assert password_field.type.of_type is str
+
+
+def test_sort_creation_fields():
+    has_default = DataclassCreationFields(
+        name="has_default",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default",
+            graphql_name="has_default",
+            default="default_str",
+            default_factory=UNSET,
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    has_default_factory = DataclassCreationFields(
+        name="has_default_factory",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="has_default_factory",
+            graphql_name="has_default_factory",
+            default=UNSET,
+            default_factory=lambda: "default_factory_str",
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    no_defaults = DataclassCreationFields(
+        name="no_defaults",
+        type_annotation=str,
+        field=StrawberryField(
+            python_name="no_defaults",
+            graphql_name="no_defaults",
+            default=UNSET,
+            default_factory=UNSET,
+            type_annotation=str,
+            description="description",
+        ),
+    )
+    fields = [has_default, has_default_factory, no_defaults]
+    assert sort_creation_fields(fields) == [
+        no_defaults,
+        has_default,
+        has_default_factory,
+    ]  # should place items with defaults last
+
+
+def test_defaults_into_factory():
+    assert (
+        defaults_into_factory(default=arguments.UNSET, default_factory=arguments.UNSET)
+        is arguments.UNSET
+    )  # should return UNSET when both defaults are UNSET
+
+    def factory_func():
+        return "strawberry"
+
+    assert (
+        defaults_into_factory(default=arguments.UNSET, default_factory=factory_func)
+        is factory_func
+    )  # should return the default_factory unchanged
+
+    mutable_default = [123, "strawberry"]
+    created_factory = defaults_into_factory(
+        default=mutable_default, default_factory=arguments.UNSET
+    )
+    assert (
+        created_factory() is not mutable_default
+    )  # should return a factory that copies the default parameter
+    assert (
+        created_factory() == mutable_default
+    )  # should return a factory that copies the default parameter
```

Comment:
```suggestion
    # should return a factory that copies the default parameter
    assert created_factory() == mutable_default
    assert created_factory() is not mutable_default
```
Nitpick: no reason to over-complicate things here